File: test/com/peeekay/aoc2023/java/Day01Test.java
package com.peeekay.aoc2023.java;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class Day01Test {
    final Day01 day = new Day01(true);

    @Test
    void part1() {
        day.solve();
        assertEquals(142, day.part1());
    }

    @Test
    void part2() {
        List<String> secondTest = List.of("""
                two1nine
                eightwothree
                abcone2threexyz
                xtwone3four
                4nineeightseven2
                zoneight234
                7pqrstsixteen
                """.split("\n"));

        day.solve(secondTest);

        assertEquals(281, day.part2());
    }
}

File: test/com/peeekay/aoc2022/kotlin/Day01Test.kt
package com.peeekay.aoc2022.kotlin

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("Day 01")
internal class Day01Test {

    @Test
    fun `Able to run against test data`() {
        val day = Day01(isTest = true)
        val answer = day.resourceAsList().first()

        assertEquals("1000", answer)
    }

    @Test
    fun `Able to run against real data`() {
        val day = Day01()
        val answer = day.resourceAsList().first()

        assertEquals("13399", answer)
    }

    @Test
    fun `Part1 with test data`() {
        val day = Day01(isTest = true)
        val answer = day.partOne()

        assertEquals(24000, answer)
    }

    @Test
    fun `Part2 with test data`() {
        val day = Day01(isTest = true)
        val answer = day.partTwo()

        assertEquals(45000, answer)
    }
}